pages/profile.php

```diff
@@ -207,53 +211,81 @@ function userProfile() {
 
-				if((health + regen) >= max_health){
-					health = max_health;
-				} else {
-					health += regen; //health ignores regen boost
-				}
-
-				if((chakra + regen) >= max_chakra){
-					chakra = max_chakra;
-				} else {
-					chakra += regen;
-				}
-
-				if((stamina + regen) >= max_stamina){
-					stamina = max_stamina;
-				} else {
-					stamina += regen;
-				}
-
-			//update health amounts / bars
-			document.getElementById('health').innerHTML = health.toFixed(2) + '/' + max_health.toFixed(2);
-			document.getElementById('healthbar').style.width = ( health / max_health )*100 + '%';
-
-			document.getElementById('chakra').innerHTML = chakra.toFixed(2) + '/' + max_chakra.toFixed(2);
-			document.getElementById('chakrabar').style.width = ( chakra / max_chakra )*100 + '%';
-
-			document.getElementById('stamina').innerHTML = stamina.toFixed(2) + '/' + max_stamina.toFixed(2);
-			document.getElementById('staminabar').style.width = ( stamina / max_stamina )*100 + '%';
+                //Check each bar to see if regen will exceed max.
+                if(statusBars.health.current + (regen * 2) > statusBars.health.max)
+                {
+                    statusBars.health.current = statusBars.health.max;
+                    statusBars.health.next_regen = statusBars.health.max;
+                }
+                else
+                {
+                    statusBars.health.current += (regen * 2);
+                    statusBars.health.next_regen = statusBars.health.current + (regen * 2);
+                }
+                
+                //Check Chakra Bar
+                if(statusBars.chakra.current + regen > statusBars.chakra.max)
+                {
+                    statusBars.chakra.current = statusBars.chakra.max;
+                    statusBars.chakra.next_regen = statusBars.chakra.max; 
+                }
+                else
+                {
+                    statusBars.chakra.current += regen ;
+                    statusBars.chakra.next_regen = statusBars.chakra.current + regen;  
+                }
+                //Check Stamina Bar
+                if(statusBars.stamina.current + regen > statusBars.stamina.max)
+                {
+                    statusBars.stamina.current = statusBars.stamina.max;
+                    statusBars.stamina.next_regen = statusBars.stamina.max; 
+                }
+                else
+                {
+                    statusBars.stamina.current += regen ;
+                    statusBars.stamina.next_regen = statusBars.stamina.current + regen;  
+                }
+                
+                //Update Health Bar
+                $('#health').html((statusBars.health.current === statusBars.health.max)? statusBars.health.current.toFixed(2) + '/' + statusBars.health.max.toFixed(2) : statusBars.health.current.toFixed(2) + '/' + statusBars.health.max.toFixed(2) + '-> <b style=\'color: green\'>' + statusBars.health.next_regen.toFixed(2) + '</b>');
```

Comment:
We ensure above that `.current` is never higher than `.max`, so I think we can remove this logic. Actually it looks like the logic above is doing the same thing regardless of whether it's maxed or not?

```suggestion
                $('#health').html(statusBars.health.current.toFixed(2) + '/' + statusBars.health.max.toFixed(2) + '-> <b style=\'color: green\'>' + statusBars.health.next_regen.toFixed(2) + '</b>');
```

Reply:
I think making that change will cause the next regen to always be displayed which we don't want. I put the check in place to prevent the regen from showing once the bar is capped. Otherwise it will display.

Reply:
Hmm I see, in that case how about we localize the ternary to the next regen preview section?
```suggestion
                let currentAndMaxHealth = statusBars.health.current.toFixed(2) + '/' + statusBars.health.max.toFixed(2) ;
                $('#health').html(currentAndMaxHealth + 
                  (statusBars.health.current === statusBars.health.max
                    ? ""
                    : '-> <b style=\'color: green\'>' + statusBars.health.next_regen.toFixed(2) + '</b>')
                );
```

Reply:
Interestingly that doesn't seem to work. 
![image](https://user-images.githubusercontent.com/10229531/186585047-38d1e258-24f6-4221-8a9e-dcb4ac9be819.png)

`//Update Health Bar
                const currentAndMaxHealth = statusBars.health.current.toFixed(2) + '/' + statusBars.health.max.toFixed(2);
                $('#health').html(currentAndMaxHealth + (statusBars.health.current === statusBars.health.max) ? '' : '-> <b style=\'color: green\'>' + statusBars.health.next_regen.toFixed(2) + '</b>');
                $('#healthbar').val(statusBars.health.current);
                
                //Update Chakra Bar
                 const currentAndMaxChakra = statusBars.chakra.current.toFixed(2) + '/' + statusBars.chakra.max.toFixed(2);
                $('#chakra').html(currentAndMaxChakra + (statusBars.chakra.current !== statusBars.chakra.max)? ('-> <b style=\'color: green\'>' + statusBars.chakra.next_regen.toFixed(2) + '</b>') : '');
                $('#chakrabar').val(statusBars.chakra.current);	
                
                //Update Stamina Bar
                const currentAndMaxStamina = statusBars.stamina.current.toFixed(2) + '/' + statusBars.stamina.max.toFixed(2);
                $('#stamina').html(currentAndMaxStamina + (statusBars.stamina.current !== statusBars.stamina.max)? ('-> <b style=\'color: green\'>' + statusBars.stamina.next_regen.toFixed(2) + '</b>') : '');
                $('#staminabar').val(statusBars.stamina.current);	`

Reply:
Ok got it working, seems calling the concat function works since it does default casting to string.